engines/director/archive.cpp

```diff
@@ -171,6 +171,18 @@ Common::String Archive::getName(uint32 tag, uint16 id) const {
 	return resMap[id].name;
 }
 
+Common::SeekableReadStreamEndian *Archive::getMovieResourceIfPresent(uint32 tag) {
+	if (g_director->getVersion() >= 400 && hasResource(tag, _movieChunks[tag]))
```

Comment:
We should check `_movieChunks.contains(tag)` before using `_movieChunks[tag]` since it's not guaranteed an chunk with a matching tag will appear in there.